2021/data/day1/example.txt

```diff
@@ -0,0 +1,10 @@
+199
+200
+208
+210
+200
+207
+240
+269
+260
+263
```

Comment:
I would love to see some unit tests with this code.

The can be pretty useful for advent, so that you can get a quick verification that you're doing the right thing with sample data.

In general, unit testing is an invaluable skill. If you're ever unsure as to what you should work on to get better as a developer, learning more about writing tests, and how to write better tests, is an incredibly valuable way to spend your time.

I personally like and use [pytest](https://docs.pytest.org/en/7.1.x/).

---

Why are tests important?

1) They ensure that the code works today
2) They ensure that the code continues to work tomorrow

It's easy to focus in on the first point, but I'd argue that the second is where 90% of the value in unit testing comes into play.

Making big sweeping changes in a well unit tested project becomes pain free, and not a scary experience.